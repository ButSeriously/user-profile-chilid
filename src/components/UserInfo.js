import PropTypes from 'prop-types';
import React from 'react';
import data from '../../storage/data.json';

class UserInfo extends React.Component {
  state = {
    user_photo: data.user_photo,
    user_name: data.user_name,
    user_location: data.user_location,
    link: data.link,
    showLink: false,
  }
  showLink = () => {
    this.setState({ showLink: !this.state.showLink });
  }
  render() {
    const showLinkClass = ['user_info__link'];
    if (this.state.showLink) {
      showLinkClass.push('user_info__link--display');
    }
    return (
      <div className="user_info">
        <div className="user_info__container">
          <div className="user_info__photo" style={{ backgroundImage: `url(${this.state.user_photo})` }} alt="user.jpg" />
          <div className="user_info__box">
            <p className="user_info__name">
              { this.state.user_name }
              <button className="user_info__like" onClick={this.props.toggleLike}>
                <i className="fa fa-heart-o" aria-hidden="true" />
              </button>
            </p>
            <p className="user_info__location">
              { this.state.user_location }
            </p>
          </div>
        </div>
        <button onClick={this.showLink} className="user_info__share">
          <i className="fa fa-share-square-o" aria-hidden="true" />
        </button>
        <div className={showLinkClass.join(' ')}>
          { this.state.link }
        </div>
      </div>
    );
  }
}

UserInfo.propTypes = {
  toggleLike: PropTypes.func.isRequired,
};

export default UserInfo;
