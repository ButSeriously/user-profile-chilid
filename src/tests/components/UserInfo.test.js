import React from 'react';
import { shallow } from 'enzyme';
import UserInfo from '../../components/UserInfo';

let toggleLike;
let wrapper;

beforeEach(() => {
  toggleLike = jest.fn();
  wrapper = shallow(<UserInfo toggleLike={toggleLike} />);
});

test('should render UserInfo correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should show share link', () => {
  wrapper.find('.user_info__share').simulate('click');
  expect(wrapper.find('.user_info__link').hasClass('user_info__link--display')).toBe(true);
});

test('should hide share link', () => {
  wrapper.setState({ showLink: true });
  wrapper.find('.user_info__share').simulate('click');
  expect(wrapper.find('.user_info__link').hasClass('user_info__link--display')).toBe(false);
});
