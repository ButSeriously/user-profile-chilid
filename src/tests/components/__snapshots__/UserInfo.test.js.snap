// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserInfo correctly 1`] = `
<div
  className="user_info"
>
  <div
    className="user_info__container"
  >
    <div
      alt="user.jpg"
      className="user_info__photo"
      style={
        Object {
          "backgroundImage": "url(./img/user-photo.jpg)",
        }
      }
    />
    <div
      className="user_info__box"
    >
      <p
        className="user_info__name"
      >
        Harvey Specter
        <button
          className="user_info__like"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="fa fa-heart-o"
          />
        </button>
      </p>
      <p
        className="user_info__location"
      >
        New York, USA
      </p>
    </div>
  </div>
  <button
    className="user_info__share"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-share-square-o"
    />
  </button>
  <div
    className="user_info__link"
  >
    https://butseriously.github.io/user-profile-chilid/
  </div>
</div>
`;
